security/advancedtls/advancedtls.go

```diff
@@ -316,6 +321,10 @@ func (c *advancedTLSCreds) ServerHandshake(rawConn net.Conn) (net.Conn, credenti
 			SecurityLevel: credentials.PrivacyAndIntegrity,
 		},
 	}
+	id := credinternal.SPIFFEIDFromState(conn.ConnectionState())
```

Comment:
If it's OK for `info.SPIFFEID` to be `nil`, this can just be `info.SPIFFEID = credinternal.SPIFFEIDFromState(conn.ConnectionState())`

Reply:
In [tls.go,](https://github.com/grpc/grpc-go/blob/master/credentials/tls.go#L126-L129) they assigned `info.SPIFFEEID` only when it's not null, but please feel free to correct me if I was wrong :)

Reply:
The code in `tls.go` can also be simplified then :)

With
```go
    id := credinternal.SPIFFEIDFromState(conn.ConnectionState())
	if id != nil {
		tlsInfo.SPIFFEID = id
	}
```

If `id == nil`, `tlsInfo.SPIFFEID` is never assigned a value, the default is `nil`
If `id != nil`, `id` is assigned to `tlsInfo.SPIFFEID`
Either way, at the end, `tlsInfo.SPIFFEID` and `id` are the same.

Reply:
That makes sense. Thanks for the clarification!